docs/architecture/adr-038-state-listening.md

```diff
@@ -222,13 +246,9 @@ type ABCIListener interface {
     ListenEndBlock(ctx types.Context, req abci.RequestEndBlock, res abci.ResponseEndBlock) error
     // ListenDeliverTx updates the steaming service with the latest DeliverTx messages
     ListenDeliverTx(ctx types.Context, req abci.RequestDeliverTx, res abci.ResponseDeliverTx) error
-    // HaltAppOnDeliveryError whether or not to halt the application when delivery of massages fails
-    // in ListenBeginBlock, ListenEndBlock, ListenDeliverTx. When `false, the app will operate in fire-and-forget mode.
-    // When `true`, the app will gracefully halt and stop the running node. Uncommitted blocks will
-    // be replayed to all listeners when the node restarts and all successful listeners that received data
-    // prior to the halt will receive duplicate data. Whether or not a listener operates in a fire-and-forget mode
-    // is determined by the listener's configuration property `halt_app_on_delivery_error = true|false`.
-    HaltAppOnDeliveryError() bool
+    // ListenCommit updates the steaming service with the latest Commit message,
+    // All the state writes of current block should have notified before this message.
+    ListenCommit(ctx types.Context, res abci.ResponseCommit) error
```

Comment:
This PR changes the semantics of ABCIListener such that _all_ implementations will HaltAppOnDeliveryError = true. This is an _enormous_ change to the behavior of Cosmos nodes which has not been motivated, analyzed, tested, or captured in the changelog.

Reply:
My assumption is that there's no existing user of this feature other than us, after all, the feature is broken in released versions, and we are the first to find it out.
I'll mention it in the changelog.